src/shell/dotnet/Shell/App.xaml.cs

```diff
@@ -175,4 +178,24 @@ private void InjectMessageRouterConfig()
                     
                 """);
     }
+   
+    private void InjectFDC3()
+    {
+        string bundle = @"fdc3-iife-bundle.js";
+        string iife = ReadResource(bundle);
+
+        WebWindow.AddPreloadScript(iife);
+    }
+
+    public string ReadResource(string name)
+    {
+        var assembly = Assembly.GetExecutingAssembly();
+        string resourcePath = $$"""Shell.Preload.{{name}}""";
```

Comment:
The behavior of this method doesn't match the name. `Shell.Preload` is specific to that particular resource (the FDC3 bundle). Instead, pass the full resource path in `name`.

Reply:
That, or rename the method to something along the lines of ReadPreloadResource